.gitignore

```diff
@@ -1,4 +1,44 @@
-/.vscode/
+# OSX trash
+.DS_Store
+
+# Eclipse files
+.classpath
+.project
+.settings/**
+
+# Files generated by JetBrains IDEs, e.g. IntelliJ IDEA
+.idea/
+*.iml
+
+# Vscode files
+.vscode
+
+# This is where the result of the go build goes
+/output*/
+/_output*/
+/_output
+
+# Emacs save files
+*~
+\#*\#
+.\#*
+
+# Vim-related files
```

Comment:
We should either (1) remove this section or (2) include a complete list of vim-related files. https://github.com/github/gitignore/blob/master/Global/Vim.gitignore

I would prefer option 1 without any vim users chiming in